r/tests/testthat/test-dplyr-funcs-datetime.R

```diff
@@ -1304,3 +1304,53 @@ test_that("dminutes, dhours, ddays, dweeks, dmonths, dyears", {
     ignore_attr = TRUE
   )
 })
+
+test_that("dseconds, dmilliseconds, dmicroseconds, dnanoseconds, dpicoseconds", {
+  example_d <- tibble(x = c(1:10, NA))
```

Comment:
Should we also test what happens when we pass floats here too? 

```
> lubridate::dseconds(1.5)
[1] "1.5s"
```

Seems to work, so we should ensure we can do that (or error helpfully if we can't for some reason)

Reply:
Of course, thanks for this! Will test and see =)

Reply:
As `duration` type in Arrow is `int64` and we can't pass floats here I will go with erroring helpfully. Will add it in the next commit.

Reply:
[ARROW-16253](https://issues.apache.org/jira/browse/ARROW-16253) might be relevant here too.

Reply:
I added a test for the error when the argument multiplied with the value of the multiplication factor of the duration helper function is float (went with easier solution - didn't go forward with forcing evaluation to check for type of an argument or try catching C++ error).